assets/js/components/dashboard-sharing/DashboardSharingSettingsButton.js

```diff
@@ -67,11 +73,18 @@ export default function DashboardSharingSettingsButton() {
 		setDialogOpen( true );
 	}, [ viewContext, hasMultipleAdmins ] );
 
-	const closeDialog = useCallback( () => {
-		setDialogOpen( false );
+	const closeDialog = useCallback(
+		( rollbackSettings = false ) => {
```

Comment:
Why do we need to have this `rollbackSettings` param? I don't think this callback is used in any other places unless I miss something.

Reply:
Uh oh, sorry for not following up here as had already spoke about this with @hussain-t on Slack DM.

@aaemnnosttv You're right. I intended to keep the `rollbackSettings` optional to not use it with `onApply`. But I didn't think about `haveSettingsChanged` turning `false` after saving changes. Thank you so much for the explanation!

Reply:
Updated to use `useEffect` to rollback settings instead of `closeDialog`. Thank you!